app/handler/provider_complaint.py

```diff
@@ -0,0 +1,233 @@
+import uuid
+from abc import ABC, abstractmethod
+from io import BytesIO
+from mailbox import Message
+from typing import Optional
+
+from app import s3
+from app.config import (
+    ALERT_COMPLAINT_REPLY_PHASE,
+    ALERT_COMPLAINT_TO_USER,
+    ALERT_COMPLAINT_FORWARD_PHASE,
+)
+from app.email import headers
+from app.email_utils import (
+    get_header_unicode,
+    parse_full_address,
+    save_email_for_debugging,
+    to_bytes,
+    render,
+    send_email_with_rate_control,
+)
+from app.log import LOG
+from app.models import (
+    User,
+    Alias,
+    DeletedAlias,
+    DomainDeletedAlias,
+    Contact,
+    ProviderComplaint,
+    Phase,
+    ProviderComplaintState,
+    RefusedEmail,
+)
+
+
+class ProviderComplaintOrigin(ABC):
```

Comment:
Nits: I'm not sure to understand the "Origin" suffix here? Maybe we can just remove it?

Reply:
Removed